src/nodeUpdate.ts

```diff
@@ -0,0 +1,17 @@
+import type { Schema, Kinds, Node, GetParams } from "./state.js"
+
+import { CanvasContext } from "./context.js"
+import { EditorAction, updateNode } from "./actions.js"
+
+export type NodeUpdateBehaviour = <S extends Schema>(node: Node<S>, action: string) => void
+
+export function makenodeUpdateBehaviour<S extends Schema>(
```

Comment:
Nit: could be camelcase similar to `makeNodeDragBehavior`. Let me know what you think?